WooCommerce/Classes/ViewRelated/Dashboard/StoreStats/StorePerformanceViewModel.swift

```diff
@@ -0,0 +1,297 @@
+        if case let .custom(_, endDate) = timeRange {
+            return endDate
+        }
+        return Date()
+    }
+
+    var buttonTitleForCustomRange: String? {
+        if case .custom = timeRange {
+            return nil
+        }
+        return Localization.addCustomRange
+    }
+}
+
+// MARK: - Private helpers
+//
+private extension StorePerformanceViewModel {
+    func observeTimeRange() {
+        $timeRange
+            .compactMap { [weak self] timeRange -> StoreStatsPeriodViewModel? in
+                guard let self else {
+                    return nil
+                }
+                return StoreStatsPeriodViewModel(siteID: siteID,
+                                                 timeRange: timeRange,
+                                                 siteTimezone: siteTimezone,
+                                                 currentDate: currentDate,
+                                                 currencyFormatter: currencyFormatter,
+                                                 currencySettings: currencySettings,
+                                                 storageManager: storageManager)
+            }
+            .sink { [weak self] viewModel in
+                guard let self else { return }
+                periodViewModel = viewModel
+                observePeriodViewModel()
+                Task { [weak self] in
+                    await self?.reloadData()
+                }
+            }
+            .store(in: &subscriptions)
+    }
+
+    func observePeriodViewModel() {
+        guard let periodViewModel else {
+            return
+        }
+
+        periodViewModel.timeRangeBarViewModel
+            .map { $0.timeRangeText }
+            .assign(to: &$timeRangeText)
+    }
+}
+
+// MARK: - Syncing data
+//
+private extension StorePerformanceViewModel {
+    @MainActor
+    func syncAllStats() async throws {
+        currentDate = Date()
+        let timezoneForSync = TimeZone.siteTimezone
+        let latestDateToInclude = timeRange.latestDate(currentDate: currentDate, siteTimezone: timezoneForSync)
+
+        try await withThrowingTaskGroup(of: Void.self) { group in
+            group.addTask { @MainActor [weak self] in
+                guard let self else { return }
+                statsVersion = await syncStats(latestDateToInclude: latestDateToInclude)
+            }
+
+            group.addTask { [weak self] in
+                try await self?.syncSiteVisitStats(latestDateToInclude: latestDateToInclude)
+            }
+
+            group.addTask { [weak self] in
+                try await self?.syncSiteSummaryStats(latestDateToInclude: latestDateToInclude)
+            }
+
+            while !group.isEmpty {
+                // rethrow any failure.
+                try await group.next()
+            }
+        }
+    }
+
+    /// Syncs store stats for dashboard UI.
+    @MainActor
+    func syncStats(latestDateToInclude: Date) async -> StatsVersion {
```

Comment:
There is a similar function in `DashboardViewModel` that does not seem to be called by anything anymore. Should it be deleted?

Reply:
This function is still being used by the legacy `DashboardViewController`. We'll have to remove it after we retire the view controller.